packages/web-components/src/components/ic-select/ic-select.tsx

```diff
@@ -316,9 +389,16 @@ export class Select {
     }
   };
 
+  private isExternalFiltering = (): boolean =>
+    this.searchable && this.disableFilter;
+
   private handleClick = (event: MouseEvent): void => {
-    this.noOptions = null;
-    this.menu.options = this.options;
+    if (this.isExternalFiltering()) {
```

Comment:
Should filteredOptions be added to options if `searchable` is true as well?

Reply:
Leaving the behaviour as is, as this would be a breaking change. If it's something requested in future we can add under a new prop